example/ndpiReader.c

```diff
@@ -2561,6 +2561,121 @@ static void debug_printf(u_int32_t protocol, void *id_struct,
 }
 #endif
 
+/**
+ * @brief From IPPROTO to string NAME
+ */
+static char* ipProto2Name(u_int16_t proto_id) {
+  static char proto[8];
+
+  switch(proto_id) {
+  case IPPROTO_TCP:
+    return("TCP");
+    break;
+  case IPPROTO_UDP:
+    return("UDP");
+    break;
+  case IPPROTO_ICMP:
+    return("ICMP");
+    break;
+  case IPPROTO_ICMPV6:
+    return("ICMPV6");
+    break;
+  case 112:
+    return("VRRP");
+    break;
+  case IPPROTO_IGMP:
+    return("IGMP");
+    break;
+  }
+
+  snprintf(proto, sizeof(proto), "%u", proto_id);
+  return(proto);
+}
+
+static int specific_protos[12] = {
+	NDPI_PROTOCOL_YOUTUBE, 
+	NDPI_PROTOCOL_YOUTUBE_UPLOAD, 
+	NDPI_PROTOCOL_TIKTOK, 
+	NDPI_PROTOCOL_GOOGLE, 
+	NDPI_PROTOCOL_GOOGLE_CLASSROOM, 
+	NDPI_PROTOCOL_GOOGLE_CLOUD, 
+	NDPI_PROTOCOL_GOOGLE_DOCS, 
+	NDPI_PROTOCOL_GOOGLE_DRIVE, 
+	NDPI_PROTOCOL_GOOGLE_MAPS, 
+	NDPI_PROTOCOL_GOOGLE_PLUS, 
+	NDPI_PROTOCOL_GOOGLE_SERVICES};
+
+int is_specific_protocol(ndpi_protocol proto)
+{
+	int ret = 0, i;
+	for (i = 0 ; i < 11 ; i ++) {
+		if (proto.app_protocol == specific_protos[i] || proto.master_protocol == specific_protos[i]) {
+			ret = 1;
+			break;
+		}
+	}
+	return ret;	
+}
+void dump_packet(struct ndpi_workflow * workflow, struct ndpi_flow_info *flow)
+{
+	/* dump packet into a file
+	*/
+	char srcip[64], dstip[64];
+	char app_name[64];
+	char stime[30], date[64];
+	struct tm tm;
+	int ret;
+	//char	szquery[512];
+
+	time_t firsttime;
+	firsttime = flow->first_seen_ms / TICK_RESOLUTION;
+	localtime_r(&firsttime, &tm);
```

Comment:
Please use something like this:

```suggestion
    struct tm result;
#ifdef WIN32
      struct tm * tm_ptr = localtime(&firsttime);
      result = *tm_ptr;
#else
      localtime_r(&firsttime, &result);
#endif
```